src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -0,0 +1,374 @@
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Native;
+import com.iab.openrtb.request.Site;
+import com.iab.openrtb.request.Video;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import org.apache.commons.collections4.CollectionUtils;
+import org.prebid.server.analytics.AnalyticsReporter;
+import org.prebid.server.analytics.model.AmpEvent;
+import org.prebid.server.analytics.model.AuctionEvent;
+import org.prebid.server.analytics.reporter.greenbids.model.CommonMessage;
+import org.prebid.server.analytics.reporter.greenbids.model.ExtBanner;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsAdUnit;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsAnalyticsProperties;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsBids;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsPrebidExt;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsSource;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsUnifiedCode;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsJacksonMapper;
+import org.prebid.server.analytics.reporter.greenbids.model.MediaTypes;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.auction.model.BidRejectionTracker;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.EncodeException;
+import org.prebid.server.log.Logger;
+import org.prebid.server.log.LoggerFactory;
+import org.prebid.server.proto.openrtb.ext.request.ExtImpPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtStoredRequest;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.NonBid;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.SeatNonBid;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.version.PrebidVersionProvider;
+import org.prebid.server.vertx.httpclient.HttpClient;
+import org.prebid.server.vertx.httpclient.model.HttpClientResponse;
+
+import java.time.Clock;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GreenbidsAnalyticsReporter implements AnalyticsReporter {
+
+    private static final String BID_REQUEST_ANALYTICS_EXTENSION_NAME = "greenbids";
+    private static final int RANGE_16_BIT_INTEGER_DIVISION_BASIS = 0x10000;
+    private static final Logger logger = LoggerFactory.getLogger(GreenbidsAnalyticsReporter.class);
```

Comment:
please add an empty line between the constants and the fields

Reply:
Logger is also a constant) usually we put it on the top

Reply:
Have moved up to the constants 👍